scripts/distribution/macOS-package/NodeConfigurationInstallerPlugin/NodeConfigurationInstallerPlugin/MyInstallerPane.m

```diff
@@ -101,17 +101,30 @@ - (void) saveConfigurationToDisk {
 }
 
 // Checks that both the mainnet and testnet node names:
-// - have length > 0
-// - contain no double quotes
+// - have length > 0 and <= 100
+// - only contain characters in [a-zA-Z0-9-_]
 - (BOOL) nodeNamesAreValid
 {
+    // NB: This character set must be accepted by the service file XML parser.
+    // This naturally excludes '<', '>', etc., but the service won't start with a sequence of ',.' in the node name either.
+    NSCharacterSet *allowedChars = [[NSCharacterSet characterSetWithCharactersInString:@"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_"] invertedSet];
```

Comment:
I would recommend adding space to this list of allowed characters since people are likely to want this.

Reply:
I allowed spaces.
However, the XML parser will ignore pre- and suffixes of spaces, so I disallowed that.